### DISCUSSION TITLE
About avrcp 1.6 "cover art" feature.

### DISCUSSION THREAD

**Discussion by planevina** [pschatzmann/ESP32-A2DP#640] [Ideas]
I made a Bluetooth speaker using ESP32 and used this library. It works very well. 

However, I have noticed that some devices can obtain cover information through avrcp1.6.
If the cover can be displayed, it would be even cooler
So I searched for relevant information and found similar support on the ESP idf website

[https://github.com/espressif/esp-idf/blob/a2f420a36ecf8cd19b443adde1d75e5eaecc6309/examples/bluetooth/bluedroid/classic_bt/a2dp_sink/main/bt_app_av.c#L473](url)

Just a discussion, is it possible for your library to add this feature? I know there are some features in ESP idf, but not in Arduino.
I don't want to use ESP idf. Your library is great, including other decoding libraries. Thank you

**Comment by pschatzmann** [comment]
Your link does not work!

If you need this functionality, why don't you extend the library and submit your change as a [pull request as described here](https://github.com/pschatzmann/ESP32-A2DP/wiki/Design-Overview) ?

You just need to [adapt these changes](https://github.com/espressif/esp-idf/commit/147a249d7a941e6a04e4e0282aaf48c55e4903c7#diff-485e81a57c0892a7d97e20c59e09f19bcde36b983385582aeb58c05617bc06e0R621)

**Reply by planevina** [reply]
Okay, thank you. I'll try transplanting it

**Comment by planevina** [comment]
I found that the condition for the cover art in the example to run properly is the latest ESP IDF. However, due to the slightly outdated version on Arduino's end, it is currently unavailable. Thank you for your reply. I will try to use it in ESP idf first

**Comment by pschatzmann** [comment]
Did you try with Arduino ESP  3.10 RC3 (IDF 5.3.1) ?

**Reply by planevina** [reply]
No. my arduino lib is 3.0.7 ,and esp idf is  5.3.1 ，It really doesn't work.
The last modification to the esp_gap-bt_api. h file was made 3 months ago, so I am currently downloading ESP IDF 5.3.2

**Comment by pschatzmann** [comment]
I just found out that Arduino Release v3.1.0 based on ESP-IDF v5.3.2 has been released 2 days ago

**Reply by planevina** [reply]
I'm sorry, I've been busy lately and forgot about this matter. The test result is: I found the explanation and examples of avrcp1.6 cover art in idf 5.4.0 and successfully ran it.   When Arduino updates to idf5.4.0, perhaps we can give it a try

**Reply by pschatzmann** [reply]
I confirm that v5.3.2 is not good enough and that it needs to be 5.4!
